### PR TITLE
Update `Suite2pSegmentationExtractor` to support multi channel and multi plane outputs

### PR DESCRIPTION
**Pull request by weiglszonja**
Related issue: #234
Also in https://github.com/catalystneuro/pinto-lab-to-nwb/issues/9

The interface PR: https://github.com/catalystneuro/neuroconv/pull/601

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
@weiglszonja Can you also include some of the fixes/improvements from https://github.com/catalystneuro/roiextractors/pull/227/files while working on this? Such as safe loading (based  on file existence) for each of the response series types

File: src/roiextractors/extractors/suite2p/suite2psegmentationextractor.py [hunk 1]
```diff
@@ -65,13 +65,13 @@ def get_available_planes(cls, folder_path: PathType):
...
         folder_path = Path(folder_path)
         prefix = "plane"
-        plane_names = []
-        for item in os.listdir(folder_path):
-            if item.startswith(prefix):
-                plane_names.append(item)
-
+        plane_paths = natsorted(folder_path.glob(pattern=prefix + "*"))
+        assert len(plane_paths), f"No planes found in '{folder_path}'."
+        plane_names = [plane_path.stem for plane_path in plane_paths]
```
Comment (weiglszonja) [inline src/roiextractors/extractors/suite2p/suite2psegmentationextractor.py:74]: @alessandratrapani  Thanks for helping with this PR! I refactored this part to make sure we are returning the name of the planes in sorted order.

**Comment by CodyCBakerPhD** [comment]
@pauladkisson @alessandratrapani Can you get your help reviewing this PR? Thanks in advance

File: tests/test_suite2psegmentationextractor.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        self.assertEqual(self.extractor.get_image_size(), self.image_size)
+
+    def test_num_frames(self):
+        self.assertEqual(self.extractor.get_num_frames(), 250)
+
+    def test_sampling_frequency(self):
+        self.assertEqual(self.extractor.get_sampling_frequency(), 10.0)
+
+    def test_channel_names(self):
+        self.assertEqual(self.extractor.get_channel_names(), ["Chan1"])
```
Comment (alessandratrapani) [inline tests/test_suite2psegmentationextractor.py:102]: I am not sure but shouldn't be ["chan1"]?

Comment (alessandratrapani) [inline tests/test_suite2psegmentationextractor.py:102]: I am not sure but shouldn't be ["chan1"]?

File: tests/test_suite2psegmentationextractor.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    def test_num_rois(self):
+        self.assertEqual(self.extractor.get_num_rois(), self.num_rois)
+
+    def test_extractor_first_channel_raw_traces(self):
+        assert_array_equal(self.extractor.get_traces(name="raw"), self.first_channel_raw_traces)
+
+    def test_extractor_second_channel(self):
+        extractor = Suite2pSegmentationExtractor(folder_path=self.folder_path, channel_name="chan2")
+        self.assertEqual(extractor.get_channel_names(), ["Chan2"])
```
Comment (alessandratrapani) [inline tests/test_suite2psegmentationextractor.py:115]: and this ["chan2]?

**Review by alessandratrapani — APPROVED** [review]
Looks good to me! :)
I just added two minor comments on the test functions for channel names. Probably I am wrong, but I wanted to be sure.

File: tests/test_suite2psegmentationextractor.py (hunk 2 above)
Comment (alessandratrapani) [reply tests/test_suite2psegmentationextractor.py:102]: Ah ok, thank you for the explanation!

**Pull request merged**